src/util/address.rs

```diff
@@ -406,6 +406,150 @@ impl Payload {
             } => script::Script::new_witness_program(version, prog)
         }
     }
+
+    /// Creates a pay to (compressed) public key hash payload from a public key
+    #[inline]
+    pub fn p2pkh(pk: &ecdsa::PublicKey) -> Payload {
+        let mut hash_engine = PubkeyHash::engine();
+        pk.write_into(&mut hash_engine)
+            .expect("engines don't error");
```

Comment:
Here and below: you can avoid these expects if use
```suggestion
        hash_engine.input(pk.serialize());
```

Reply:
Or even simpler: `PubkeyHash::hash(pk.serialize())`

Reply:
Very nice! Added a new commit.